### PR TITLE
refactor: add `Validate()` and `Check()`

### PR DESCRIPTION
**Pull request by zyy17**
## Summary by CodeRabbit

- **New Features**
	- Introduced validation methods for `GreptimeDBCluster` and `GreptimeDBStandalone`, ensuring configurations are correct and resources are available.
	- Added checks for TLS secrets and object storage provider credentials.
  
- **Bug Fixes**
	- Enhanced error handling in the `Reconcile` method by logging warnings for invalid configurations.

- **Refactor**
	- Streamlined validation logic by removing outdated methods and consolidating checks within the new validation structure.

### CODE REVIEW

File: apis/v1alpha1/validate.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+}
+
+// Validate checks the GreptimeDBStandalone and returns an error if it is invalid.
+func (in *GreptimeDBStandalone) Validate() error {
+	if in == nil {
+		return nil
+	}
+
+	if err := validateTomlConfig(in.GetConfig()); err != nil {
+		return fmt.Errorf("invalid toml config: '%v'", err)
```
Comment (daviderli614) [inline apis/v1alpha1/validate.go:146]:
```suggestion
		return fmt.Errorf("invalid standalone toml config: '%v'", err)
```

**Comment by daviderli614** [comment]
Can you add a `validate_test.go` file to test some validity?

**Comment by zyy17** [comment]
> Can you add a `validate_test.go` file to test some validity?

done

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**